alica_engine/include/engine/blackboard/Blackboard.h

```diff
@@ -308,22 +313,30 @@ class BlackboardImpl
         };
     };
 
-    // trait to check if a type is one of the types in the variant
-    template <class T1, class T2>
-    struct isTypeInVariant;
+    // trait to find the index of the given type in the variant, -1 if not found
+    template <class T, class Variant, std::size_t numTypes>
+    struct findInVariantImpl;
 
-    template <class T>
-    struct isTypeInVariant<T, std::variant<>> : std::false_type
+    template <class T, std::size_t NUM_TYPES>
+    struct findInVariantImpl<T, std::variant<>, NUM_TYPES> : std::integral_constant<std::ptrdiff_t, -1>
```

Comment:
perhaps with (constexp) functions it could be more readable. But I guess it's more about personal preferences

Reply:
Not sure if I can do using constexpr. Would be nice if you could implement & I can learn as well